src/store/portfolioSummary/index.ts

```diff
@@ -0,0 +1,256 @@
+/* eslint-disable camelcase */
+import {PortfolioSummaryDTO} from "@/api/models";
+import {Action, getModule, Module, Mutation, VuexModule} from "vuex-module-decorators";
+import rootStore from "@/store";
+import {nameofProperty, retrieveSession, storeDataToSession} from "@/store/helpers";
+import Vue from "vue";
+import {api} from "@/api";
+import {AxiosRequestConfig} from "axios";
+
+const ATAT_PORTFOLIO_SUMMARY_KEY = "ATAT_PORTFOLIO_SUMMARY_KEY";
+
+@Module({
+  name: "PortfolioSummaryStore",
+  namespaced: true,
+  dynamic: true,
+  store: rootStore,
+})
+export class PortfolioSummaryStore extends VuexModule {
+  initialized = false;
+  portfolioSummaryList: PortfolioSummaryDTO[] | null = null;
+
+  @Action
+  public async getAllPortfolioSummaryList(): Promise<PortfolioSummaryDTO[] | null> {
+    return this.portfolioSummaryList;
+  }
+
+  // store session properties
+  protected sessionProperties: string[] = [
+    nameofProperty(this, (x) => x.portfolioSummaryList)
+  ];
+
+  @Mutation
+  public setStoreData(sessionData: string): void {
+    try {
+      const sessionDataObject = JSON.parse(sessionData);
+      Object.keys(sessionDataObject).forEach((property) => {
+        Vue.set(this, property, sessionDataObject[property]);
+      });
+    } catch (error) {
+      throw new Error("error restoring session for portfolio summary data store");
+    }
+  }
+
+  @Mutation
+  public setInitialized(value: boolean): void {
+    this.initialized = value;
+  }
+
+  @Mutation
+  public setPortfolioSummaryList(value: PortfolioSummaryDTO[]): void {
+    this.portfolioSummaryList = value;
+    storeDataToSession(
+      this,
+      this.sessionProperties,
+      ATAT_PORTFOLIO_SUMMARY_KEY
+    );
+  }
+
+  @Action({rawError: true})
+  async initialize(): Promise<void> {
+    const sessionRestored = retrieveSession(ATAT_PORTFOLIO_SUMMARY_KEY);
+    if (sessionRestored) {
+      this.setStoreData(sessionRestored);
+      this.setInitialized(true);
+    }
+  }
+
+  @Action({rawError: true})
+  async ensureInitialized(): Promise<void> {
+    await this.initialize();
+  }
+
+  /**
+   * Constructs a single query that gets all the CSP records across all the portfolis. Parses
+   * the response and sets the 'csp_display' to the respective portfolio.
+   */
+  @Action({rawError: true})
+  private async setCspDisplay(portfolioSummaryList: PortfolioSummaryDTO[]) {
+    const cspSysIds = portfolioSummaryList.map(portfolio => portfolio.csp.value);
+    const allCspList = await api.cloudServiceProviderTable.getQuery(
+      {
+        params:
+          {
+            sysparm_fields: 'sys_id,name',
+            sysparm_query: `sys_idIN` + cspSysIds
```

Comment:
just as an FYI, in general, we stick to double-quotes in our js unless there's a need for ticks or single quotes

Reply:
In my last project we had a linter in place that would complain about double quote usage in typescript files. Just became a habit. But will change to double quotes.